client/src/components/WaitingRoom/index.js

```diff
@@ -0,0 +1,99 @@
+import React, { Component } from 'react';
+import { withFirebase } from '../Firebase';
+import { compose } from 'recompose';
+import { withAuthorization } from '../Session';
+
+class WaitingRoomBase extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      players: [],
+      gameId: '',
+      timeLimit: 10,
+      started: false
+    };
+
+    this.onPlayerListUpdate = this.onPlayerListUpdate.bind(this);
+    this.onGameStart = this.onGameStart.bind(this);
+    this.startGame = this.startGame.bind(this);
+  }
+
+  async componentDidMount() {
+    //Grab gameId & timeLimit to set those as well as gameName
+    const { id } = this.props.match.params;
+    const gameInstance = await this.props.firebase.game(id).get();
+    this.setState({
+      gameId: id,
+      timeLimit: gameInstance.data().timeLimit,
+      started: gameInstance.data().hasStarted,
+      players: gameInstance.data().players
+    });
+  }
+
+
+
+  //Listens for an update to players
+  async onPlayerListUpdate(gameId) {
+    const game = this.props.firebase.game(gameId).get();
+    game.on('child_added', async function(snapshot)
```

Comment:
Does this actually work? From what I found, this was supposed to be syntax for Firebase realtime database and not Firestore https://stackoverflow.com/questions/54479892/difference-between-get-and-snapshot-in-cloud-firestore

Reply:
The Firestore syntax should be `onSnapshot()` for adding a listener. Also this function is never called so this code never executes. If you want a listener, it should be called or added in `componentDidMount()` and detached in `componentWillUnmount()`.

Reply:
deleted and added to componentDidMount method

Reply:
Have you pushed the changes? I'm not seeing it on the PR.

Reply:
just pushed, sorry i was trying to address the commit question